File: general/compact_object.py
def compactObject(obj):
    final = [] if type(obj) == list else {}
    
    if (type(obj) == list):
        for x in obj:
            if (type(x) == list or type(x) == dict):
                final.append(compactObject(x))
            else:
                if bool(x): final.append(x)
    else:
        for i, j in obj.items():
            if (type(j) == list or type(j) == dict):
                final[i] = compactObject(j)
            else:
                if bool(j): final[i] = j
            
    return final


# Run the function
if __name__ == "__main__":
    obj = {"a": None, "b": [False, 1]}
    print(compactObject(obj))  # {'b': [1]}

File: leetcode/30Days_JS/compact_object.js
/**
 * @param {Object|Array} obj
 * @return {Object|Array}
 */
var compactObject = function(obj) {
    const final = Array.isArray(obj) ? [] : {};

    if (Array.isArray(obj)) {
        obj.forEach(value => {
            if (typeof value === 'object' && Boolean(value)){
                final.push(compactObject(value))
            } else {
                if (Boolean(value))
                final.push(value)
            }
        })
    } else {
        for (let prop of Object.entries(obj)){
            if (typeof prop[1] === 'object' && Boolean(prop[1])){
                final[prop[0]] = compactObject(prop[1])
            } else {
                if (Boolean(prop[1]))
                final[prop[0]] = prop[1]
            }
        }
    }
    return final
};